lib/pbench/server/api/resources/__init__.py

```diff
@@ -274,13 +273,19 @@ def convert_json(value: JSON, parameter: "Parameter") -> JSON:
     try:
         if json.loads(json.dumps(value)) == value:
             if parameter.keywords:
-                bad = set(value.keys()) - set(parameter.keywords)
+                bad = []
+                for k in value.keys():
+                    if (
+                        k not in parameter.keywords
+                        and f"{k.split('.')[0]}." not in parameter.keywords
+                    ):
+                        bad.append(k)
                 if bad:
                     raise KeywordError(parameter, "JSON keys", bad)
```

Comment:
While it would be possible to do the transformation as a `map`ping, that's not it: we need to check the full original path as well as the root term with the `.` (now `.*`) suffix; plus, I want to report the original bad term (e.g., `user.contact`) rather than the root `user.*` term if it doesn't match. I think this would end up being messier and more complicated than what I wrote.